packages/playground-preview-worker/tests/index.test.ts

```diff
@@ -0,0 +1,141 @@
+import { unstable_dev } from "wrangler";
+import type { UnstableDevWorker } from "wrangler";
+
+function removeUUID(str: string) {
+	return str.replace(
+		/\w{8}-\w{4}-\w{4}-\w{4}-\w{12}/g,
+		"00000000-0000-0000-0000-000000000000"
+	);
+}
+
+const REMOTE = "https://playground-testing.devprod.cloudflare.dev";
+const PREVIEW_REMOTE =
+	"https://random-data.playground-testing.devprod.cloudflare.dev";
+
+async function fetchUserToken() {
+	return fetch(REMOTE).then(
+		(r) => r.headers.getSetCookie()[0].split(";")[0].split("=")[1]
+	);
+}
+describe("Preview Worker", () => {
+	let defaultUserToken: string;
+	beforeAll(async () => {
+		defaultUserToken = await fetchUserToken();
+
+		await fetch(`${REMOTE}/api/worker`, {
+			method: "POST",
+			headers: {
+				cookie: `user=${defaultUserToken}`,
+				"Content-Type":
+					"multipart/form-data; boundary=----WebKitFormBoundaryqJEYLXuUiiZQHgvf",
+			},
+			body: '------WebKitFormBoundaryqJEYLXuUiiZQHgvf\nContent-Disposition: form-data; name="index.js"; filename="index.js"\nContent-Type: application/javascript+module\n\nexport default {\n	fetch(request) {\n		const url = new URL(request.url)\n		if(url.pathname === "/exchange") {\n			return Response.json({\n				token: "TEST_TOKEN",\n				prewarm: "TEST_PREWARM"\n			})\n		}\n		if(url.pathname === "/redirect") {\n			return Response.redirect("https://example.com", 302)\n		}\n		if(url.pathname === "/method") {\n			return new Response(request.method)\n		}\n		if(url.pathname === "/status") {\n			return new Response(407)\n		}\n		if(url.pathname === "/header") {\n			return new Response(request.headers.get("X-Custom-Header"))\n		}\n		return Response.json({\n			url: request.url,\n			headers: [...request.headers.entries()]\n		})\n	}\n}\n\n------WebKitFormBoundaryqJEYLXuUiiZQHgvf\nContent-Disposition: form-data; name="metadata"; filename="blob"\nContent-Type: application/json\n\n{"compatibility_date":"2023-05-04","main_module":"index.js"}\n------WebKitFormBoundaryqJEYLXuUiiZQHgvf--',
+		}).then((response) => response.json());
+	});
+
+	it("should be redirected with cookie", async () => {
+		const resp = await fetch(
+			`${PREVIEW_REMOTE}/.update-preview-token?token=TEST_TOKEN&suffix=${encodeURIComponent(
+				"/hello?world"
+			)}`,
+			{
+				method: "GET",
+				redirect: "manual",
+			}
+		);
+		expect(resp.headers.get("location")).toMatchInlineSnapshot(
+			'"/hello?world"'
+		);
+		expect(resp.headers.get("set-cookie") ?? "").toMatchInlineSnapshot(
+			'"token=TEST_TOKEN; Domain=random-data.playground-testing.devprod.cloudflare.dev; HttpOnly; Secure; SameSite=None"'
+		);
+	});
+
+	it("should convert cookie to header", async () => {
+		const resp = await fetch(PREVIEW_REMOTE, {
+			method: "GET",
+			headers: {
+				cookie: `token=${defaultUserToken}`,
+			},
+		});
+
+		const json = (await resp.json()) as { headers: string[][]; url: string };
+
+		expect(
+			json.headers.find((h) => h[0] === "cf-workers-preview-token")
+		).toMatchInlineSnapshot("undefined");
```

Comment:
is this snapshot the correct expected value?

Reply:
Probably not... Thanks for catching this!